### PR TITLE
feat/icon component

### PR DESCRIPTION
**Pull request by HellraiserCW**

### CODE REVIEW

File: src/app/shared/components/svg-icon/svg-icon.component.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+@Component({
+  selector: 'epm-svg-icon',
+  standalone: true,
+  imports: [CommonModule, NgOptimizedImage],
+  templateUrl: './svg-icon.component.html',
+  styleUrls: ['./svg-icon.component.scss']
+})
+export class SvgIconComponent {
+  @Input() svgIconName: SOCIAL_ICONS = SOCIAL_ICONS.none;
```
Comment (GennadiyShpak) [inline src/app/shared/components/svg-icon/svg-icon.component.ts:13]:
```suggestion
  @Input({required: true}) svgIconName: SOCIAL_ICONS = SOCIAL_ICONS.none;
```

File: src/app/app.config.ts [hunk 2]
```diff
@@ -15,12 +15,12 @@ export enum APP_ROUTER_NAME {
...
-  Facebook = 'assets/icons/facebook_24x24.png',
-  Skype = 'assets/icons/skype_24x24.png',
-  LinkedIn = 'assets/icons/linkedin_24x24.png',
-  Telegram = 'assets/icons/telegram_24x24.png',
+  Teams = 'assets/icons/social-icons/teams.svg',
+  Instagram = 'assets/icons/social-icons/instagram.svg',
+  Facebook = 'assets/icons/social-icons/facebook.svg',
+  Skype = 'assets/icons/social-icons/skype.svg',
+  LinkedIn = 'assets/icons/social-icons/linkedin.svg',
+  Telegram = 'assets/icons/social-icons/telegram.svg',
```
Comment (GennadiyShpak) [inline src/app/app.config.ts:23]: why do we need full path? can we avoid it?

File: src/app/shared/components/svg-icon/svg-icon.component.ts (hunk 1 above)
Comment (HellraiserCW) [reply src/app/shared/components/svg-icon/svg-icon.component.ts]: Good idea, thanks, updated my knowledge about @Input in Angular 16

File: src/app/app.config.ts (hunk 2 above)
Comment (HellraiserCW) [reply src/app/app.config.ts:23]: I think in this case we need a simple, centralized solution, so storing the full path in an enum works good.
Do we need to use getter in component like that:
   get iconPath() {
    return `assets/icons/social-icons/${this.svgIconName}.svg`;
  }
  or set the path to icon in local property in lifecycle hook?

Comment (GennadiyShpak) [reply src/app/app.config.ts:23]: why do we need getter?

Comment (HellraiserCW) [reply src/app/app.config.ts]: Moved path from config to component's template literal.

File: src/app/shared/components/svg-icon/svg-icon.component.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<img
+  [ngSrc]="'assets/icons/social-icons/' + svgIconName + '.svg'"
+  width="24"
+  height="24"
+  [alt]="svgIconName + ' icon'"
```
Comment (GennadiyShpak) [inline src/app/shared/components/svg-icon/svg-icon.component.html:5]: please use template string

Comment (HellraiserCW) [reply src/app/shared/components/svg-icon/svg-icon.component.html]: fixed

**Review by GennadiyShpak — APPROVED** [review]

**Pull request merged**